pkg/chartRepo/ChartRepositoryService.go

```diff
@@ -172,13 +172,15 @@ func (impl *ChartRepositoryServiceImpl) CreateChartRepo(request *ChartRepoDto) (
 			if err != nil {
 				return nil, err
 			}
-			data, err := impl.updateRepoData(cm.Data, request)
-			if err != nil {
-				impl.logger.Warnw(" config map update failed", "err", err)
-				continue
+			if cm.Data != nil {
+				data, err := impl.updateRepoData(cm.Data, request)
+				if err != nil {
```

Comment:
propagate the error and rollback any other modifications

Reply:
propagate all the errors and rollback changes if any

Reply:
have changed the code here, secrets will get created in case of public as well as private, also db updation is happening in transaction so rollback will happen in case of errors.